pkg/engine/shield/remove.go

```diff
@@ -11,7 +11,17 @@ func (mgr *Manager) RemoveShield(id key.Shield, target key.TargetID) {
 		return
 	}
 
-	delete(mgr.targets[target], id)
+	i := 0
+	var removedShield []*Instance
+	for _, shield := range mgr.targets[target] {
+		if shield.name == id {
+			removedShield = append(removedShield, shield)
+		} else {
+			mgr.targets[target][i] = shield
+			i++
+		}
+	}
```

Comment:
Delete `removedShield` since it is unused (should only be necessary if you are removing multiple shields at once).